src/MCPServer/lib/archivematicaMCP.py

```diff
@@ -138,57 +144,50 @@ def findOrCreateSipInDB(path, waitSleep=dbWaitSleep, unit_type='SIP'):
 
 @log_exceptions
 @auto_close_db
-def createUnitAndJobChain(path, config, terminate=False):
+def createUnitAndJobChain(path, watched_dir, workflow):
     path = unicodeToStr(path)
     if os.path.isdir(path):
         path = path + "/"
-    logger.debug('Creating unit and job chain for %s with %s', path, config)
+    logger.debug('Starting chain for %s', path)
+    if not os.path.exists(path):
+        return
     unit = None
+    unit_type = watched_dir["unit_type"]
     if os.path.isdir(path):
-        if config[3] == "SIP":
+        if unit_type == "SIP":
             UUID = findOrCreateSipInDB(path)
             unit = unitSIP(path, UUID)
-        elif config[3] == "DIP":
+        elif unit_type == "DIP":
             UUID = findOrCreateSipInDB(path, unit_type='DIP')
             unit = unitDIP(path, UUID)
-        elif config[3] == "Transfer":
+        elif unit_type == "Transfer":
             unit = unitTransfer(path)
     elif os.path.isfile(path):
-        if config[3] == "Transfer":
+        if unit_type == "Transfer":
             unit = unitTransfer(path)
         else:
             return
             UUID = uuid.uuid4()
             unit = unitFile(path, UUID)
```

Comment:
Related to my comments [here](https://github.com/artefactual/archivematica/pull/1251#discussion_r246364008). The scope here is well-known enough, and tight enough (part of the same decision-path you're modifying), to change as part of this PR given normal circumstances. Given my initial question, this would also make it easier on the future reader.

Reply:
Okay, will do.

Reply:
I've removed it.